vendor_data.py

```diff
@@ -56,101 +52,190 @@ def __init__(self, VendorName, TargLang, WordRate, VendorMail = "", CatTool = "X
+            if done.lower() == "yes":
+                Vndr=VendorData(ProjectName, SourceLang, VendorName, TargLang, WordRate, Preferred, VendorMail, CatTool)
+                Excel = pyip.inputStr("Do you want add this vendor to the excel file? ")
+                if Excel.lower() == "yes":
+                    Vndr.to_excel()
+    if args.modify == True:
+        done = "no"
+        while done.lower() != "yes":
+            ProjectName = pyip.inputStr("What is the name of the project that the vendor you want to modify works on? ")
+            SourceLang = pyip.inputStr("What is the source language? ")
+            filename = "_".join([str(ProjectName), str(SourceLang)]) + ".xlsx"
+            if os.path.exists(filename):
+                excel_records = pd.read_excel(filename)
+                excel_records_df = excel_records.loc[:, ~excel_records.columns.str.contains('^Unnamed')]
+                VendorDict=excel_records_df.to_dict()
+
+                vendors = VendorDict["Vendor"]
+                print ("These are the vendor's already in the database: ")
+                for index, vendor_name in vendors.items():
+                    print(f"{index}: {vendor_name}")
+                
+                vendor_index = pyip.inputInt("Enter the index of the vendor you want to modify: ")
+                if vendor_index in vendors:
+                    vendor_key = "Vendor"
+                    email_key = "E-mail"
+                    cat_tool_key = "CAT Tool"
+                    word_rate_key = "Word Rate"
+                    status_key = "Status"
+
+                    new_email = pyip.inputStr("What is the vendor's e-mail? ", blank = True)
+                    new_cat_tool = pyip.inputMenu(["XTM", "Trados Studio", "MemoQ", "Memsource"], prompt = "In which tool will the vendor be working?", blank = True, default = "XTM")
+                    new_word_rate = pyip.inputNum("What is the vendor's word rate? Should be between 0.01 and 0.15. If higher, choose another vendor. ", blank = True)
+                    new_status = pyip.inputMenu(["Potential", "Preferred", "Back-up"], prompt = "What is the vendor's new status? ", blank = True)
+                    done = pyip.inputStr("Are you done? ")
+            if done.lower() == "yes":
+
+                    if new_email:
+                        change_dictionary_value(VendorDict, email_key, vendor_index, new_email)
+                    if new_cat_tool:
+                        change_dictionary_value(VendorDict, cat_tool_key, vendor_index, new_cat_tool)
+                    if new_word_rate:
+                        change_dictionary_value(VendorDict, word_rate_key, vendor_index, new_word_rate)
+                    if new_status:
+                        change_dictionary_value(VendorDict, status_key, vendor_index, new_status)
+                    df = pd.DataFrame(VendorDict)
+                    with pd.ExcelWriter(filename, engine="openpyxl", mode="a", if_sheet_exists="replace") as writer:

```

Comment:
You do mode "a" instead of mode "w" in case there is information in another sheet?